### PR TITLE
Add strategy filter

### PR DESCRIPTION
**Pull request by itkg-alavieille**
https://github.com/open-orchestra/open-orchestra-libs/pull/11
https://github.com/open-orchestra/open-orchestra-user-bundle/pull/71
https://github.com/open-orchestra/open-orchestra-workflow-function-bundle/pull/39
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1019
https://github.com/open-orchestra/open-orchestra-base-api-mongo-model-bundle/pull/6
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/406
https://github.com/open-orchestra/open-orchestra-base-api-bundle/pull/49
https://github.com/open-orchestra/open-orchestra-base-bundle/pull/35

### CODE REVIEW

File: BaseBundle/DependencyInjection/OpenOrchestraBaseExtension.php [hunk 1]
```diff
@@ -32,5 +32,11 @@ public function load(array $configs, ContainerBuilder $container)
 
         $container->setParameter('open_orchestra_base.administration_languages', $config['administration_languages']);
         $container->setParameter('open_orchestra_base.encryption_key', $config['encryption_key']);
+
+        $filterStrategyFolder = $container->getParameter("kernel.root_dir")."/../".$container->getParameter('open_orchestra_filter_type_strategy_folder');
```
Comment (nicolasThal) [inline BaseBundle/DependencyInjection/OpenOrchestraBaseExtension.php]: the parameter should not be found like that.

You should add a configuration in the `Configuration.php` file which will be used after for the directory

File: BaseBundle/DependencyInjection/OpenOrchestraBaseExtension.php [hunk 2]
```diff
@@ -32,5 +32,11 @@ public function load(array $configs, ContainerBuilder $container)
...
+        if (is_dir($filterStrategyFolder)) {
+            $loaderPaginationConfig = new Loader\YamlFileLoader($container, new FileLocator($filterStrategyFolder));
+            $loaderPaginationConfig->load($container->getParameter("open_orchestra_filter_type_strategy_config"));
```
Comment (nicolasThal) [inline BaseBundle/DependencyInjection/OpenOrchestraBaseExtension.php]: same

File: BaseBundle/Resources/config/parameters.yml [hunk 3]
```diff
@@ -1,2 +1,4 @@
 parameters:
     open_orchestra_base.preview_route_pattern: /preview/{nodeId}/{aliasId}/{token}
+    open_orchestra_filter_type_strategy_folder: vendor/open-orchestra/open-orchestra-libs/OpenOrchestra/Pagination/MongoTrait/FilterTypeStrategy/
```
Comment (nicolasThal) [inline BaseBundle/Resources/config/parameters.yml]: should be in the bundle `Configuration.php` file

File: phpunit.xml.dist [hunk 4]
```diff
@@ -16,6 +16,7 @@
     <testsuites>
         <testsuite name="unit">
             <directory>*Bundle/Tests</directory>
+            <exclude>./BaseBundle/Tests/Functional</exclude>
```
Comment (nicolasThal) [inline phpunit.xml.dist]: Do not forget to add such  aline in the `open-orchestra` projects

**Comment by itkg-alavieille** [comment]
@nicolasThal

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**